Makefile

```diff
@@ -0,0 +1,96 @@
+ROOT_DIR := $(shell git rev-parse --show-toplevel)
+VERSION := $(shell grep version $(ROOT_DIR)/package.json | cut -d':' -f2 | tr -d "\"', \r")
+PACKAGE_NAME := grafana-synthetic-monitoring-app-$(VERSION).zip
+PACKAGE_NAME_LATEST := grafana-synthetic-monitoring-app-latest.zip
+ARTIFACTS_DIR ?= $(ROOT_DIR)/artifacts/builds
+PROM_VERSION := 2.48.1
+
+.PHONY: build
+build:
+	yarn build
+
+# requires GCP_KEY
+# we need to authenticate to private NPM registry before running yarn install
+# .PHONY: authenticate-npm
+# authenticate-npm:
+# ifdef GCP_KEY
+# 	printenv GCP_KEY > /tmp/gcpkey.json
+# 	export GOOGLE_APPLICATION_CREDENTIALS=/tmp/gcpkey.json && yarn authenticate-npm
+# 	rm /tmp/gcpkey.json
+# else 
+# 	yarn authenticate-npm
+# endif
+
+.PHONY: install
+install:
+	yarn install
+
+.PHONY: lint
+lint:
+	yarn lint
+
+# requires GRAFANA_API_KEY
+.PHONY: sign
+sign:
+	yarn sign
+
+.PHONY: test
+test:  # install-promql-deps test-promql
+	yarn test
+
+# yarn tsc prints a lot of errors even if queries.js is generated successfully,
+# so drop stdout (keeping stderr) and do an explicit test to see if the file
+# was created.
+scripts/node-generate/queries.js: src/queries.ts
+	rm -f scripts/node-generate/queries.js ;\
+	yarn tsc --outDir scripts/node-generate --pretty false src/queries.ts >/dev/null ;\
+	test -f scripts/node-generate/queries.js && echo 'Ignore error above, file generated successfully'
+
+rules.json: scripts/node-generate/queries.js
+	node scripts/node-generate/node-generate.js
+
+rules.yml: rules.json
+	if yq -h | grep "jq wrapper for YAML documents" > /dev/null; then \
+  		yq -y . rules.json > rules.yml; \
+  	else \
+		yq -Poy rules.json > rules.yml; \
+	fi
+	rm rules.json
+
+test-promql: rules.yml
+	promtool check rules --lint=all --lint-fatal rules.yml
+
+.PHONY: install-promql-deps
+install-promql-deps:
```

Comment:
All the promql rules are for something that will come later. The k8s crew has a really interesting system of extracting promql queries and validating them as part of CI